src/python/WMCore/MicroService/Service/Data.py

```diff
@@ -94,6 +95,11 @@ def get(self, **kwds):
             res.update(self.mgr.status(detail, **kwds))
         elif kwds.get('API') == "info":
             res.update(self.mgr.info(kwds.pop("request", None), **kwds))
+        if cherrypy.request.headers.get('Accept-Encoding', '') == 'gzip':
+            cherrypy.response.headers['Content-Encoding'] = 'gzip'
+            # generate data similar to restcall, e.g. `{"result":[...]}`
+            data = {"result": results(res)}
```

Comment:
If we return compressed data, then I would expect the whole object to be compressed, not a compressed object inside a json object. In other words, I think even the `{'result':...}` should be compressed in the response object.

Reply:
@amaltaro , if I recall correctly we can't use `cherrypy.tools.gzip` decorator because we use WMCore JSONFormatter which transforms data into generator, in other words you can't gzip generator. I think I tried how to refactor a code but have not found a general way to satisfy both use cases.

Meanwhile, I do not understand your comment either, you said *I think even the* `{'result':...}` *should be compressed*. This is exactly what next line is doing it compress entire data, see next line which states `gzip.compress(bytes(json.dumps(data), 'utf-8'))`.

Reply:
Sorry, you are right!